### ISSUE TITLE
Exporting Opentelemetry for few metrics

### ISSUE THREAD

**Issue by AshwinKul28** [DiceDB/dice#307]
This issue is to export opentelemetry metrics to the supported providers [Jaeger](https://jaegertracing.io/), [Zipkin](https://zipkin.io/), [Prometheus](https://prometheus.io/), or a [vendor-specific](https://opentelemetry.io/ecosystem/vendors/) backend.

This issue is a dependent on #306

**Comment by AshwinKul28** [comment]
@JyotinderSingh Please assign this issue to me and @gauravsarma1992

**Comment by JyotinderSingh** [comment]
> @JyotinderSingh Please assign this issue to me and @gauravsarma1992 

Done. @gauravsarma1992 please comment here and I'll be able to assign the issue to you as well.

**Comment by gauravsarma1992** [comment]
@JyotinderSingh Please assign it to me as well.

**Comment by soumya-codes** [comment]
@AshwinKul28 @gauravsarma1992 if this task involves coming up with the list of metrics that we plan to export, I would suggest you document the list matrices we want to export and put it up for review.

**Comment by gauravsarma1992** [comment]
@soumya-codes This task is to mainly export few metrics to provide as an example for future PRs. This will serve as a baseline for all other metrics. Once we have this, we can open up the collection, calculation and propagation of metrics to the entire team.

**Comment by soumya-codes** [comment]
@gauravsarma1992 thanks, now i understand. 
IMO it would be great if we can arrive at the list of important metrices that would be most useful and add them make them part of the baseline. So while you both are implementing the framework, in parallel, it may be a good thing to come with the list of initial useful  5-10 metrices or what ever is the minimum necessary number and add them.

Just an opinion, please take the approach that suits you best and enables you move faster.

**Comment by gauravsarma1992** [comment]
Yes, that makes sense. For Metrics collection, these are the following phases:
1. Introduce the Opentelemetry SDK into DiceDB
2. Integrate with a Opentelemetry provider like Prometheus or Signoz
3. Create baseline PRs for internal and external metrics for DiceDB
4. Open up the metrics floodgate for the wider team

Once we are past step 1 and 2, let's recoup to define the list of the important metrics that can be exported by the team.
cc @AshwinKul28

**Comment by arpitbbhayani** [comment]
Hello @AshwinKul28,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by arpitbbhayani** [comment]
Hello @arpitbbhayani,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by arpitbbhayani** [comment]
Hello @arpitbbhayani,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by arpitbbhayani** [comment]
Hello @arpitbbhayani,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.
